### PR TITLE
Documentation: Added InstancedMesh page.

### LINKED ISSUE #17621: Lack of InstancedMesh document and TS definition
##### Description of the problem

From #17505. In r109 we have `InstancedMesh`. But no the document and TS definition yet. Let's add them in r110.

##### Three.js version

- [x] Dev
- [ ] r109
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

### PR DESCRIPTION
**Pull request by Mugen87**
Fixed #17621

### CODE REVIEW

**Comment by Mugen87** [comment]
https://raw.githack.com/Mugen87/three.js/dev38/docs/index.html#api/en/objects/InstancedMesh

**Comment by donmccurdy** [comment]
I was surprised at not seeing InstancedBufferGeometry or InstancedBufferAttribute mentioned here. From the example I can see they aren't required to use InstancedMesh, but should they be mentioned or recommended? Does InstancedMesh support them?

File: docs/api/en/objects/InstancedMesh.html [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+		<script src="page.js"></script>
+		<link type="text/css" rel="stylesheet" href="page.css" />
+	</head>
+	<body>
+		[page:Mesh] &rarr;
+
+		<h1>[name]</h1>
+
+		<p class="desc">
+		A special version of [page:Mesh] with instanced rendering support. Use [name] if you have to render a large amount of
```
Comment (donmccurdy) [inline docs/api/en/objects/InstancedMesh.html]: Nit: Here, and throughout the rest of the page, I would replace the word "amount" with "number". The latter is preferred for objects that can be counted discretely. (https://www.dailywritingtips.com/amount-vs-number/)

**Comment by mrdoob** [comment]
> I was surprised at not seeing InstancedBufferGeometry or InstancedBufferAttribute mentioned here. From the example I can see they aren't required to use InstancedMesh, but should they be mentioned or recommended? Does InstancedMesh support them?

I don't know yet.

I personally do not like the `InstancedBufferGeometry`/`InstancedBufferAttribute` API but it allows devs do things that `InstancedMesh` currently doesn't support.

File: docs/api/en/objects/InstancedMesh.html [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+		<p>
+			The amount of instances. The *count* value passed into the constructor represents the maximum amount of
+			instances of this mesh. You can change the amount of instances at runtime to an integer value
+			in the range [0, count].
+		</p>
+		<p>
+			If you need more instances than the original count value, you have to create a new [name].
+		</p>
+
+		<h3>[property:Boolean isInstancedMesh]</h3>
```
Comment (donmccurdy) [inline docs/api/en/objects/InstancedMesh.html]: Not anything to fix in this PR, but maybe we should have a "readonly" tag for this and other similar cases in the API. 🤔

**Comment by mrdoob** [comment]
Thanks!

File: docs/api/en/objects/InstancedMesh.html (hunk 2 above)
Comment (mrdoob) [reply docs/api/en/objects/InstancedMesh.html]: I don't think we should document them actually...

File: docs/api/en/objects/InstancedMesh.html (hunk 1 above)
Comment (Mugen87) [reply docs/api/en/objects/InstancedMesh.html]: Ok, let me make a new PR with your suggested change 👍

File: docs/api/en/objects/InstancedMesh.html (hunk 2 above)
Comment (Mugen87) [reply docs/api/en/objects/InstancedMesh.html]: Then let me remove it^^

**Pull request merged**